docs/manual/java/guide/cluster/UsingAkkaPersistenceTyped.md

```diff
@@ -0,0 +1,185 @@
+# Domain Modelling with Akka Persistence Typed
+
+This section presents all the steps to model an [Aggregate](https://martinfowler.com/bliki/DDD_Aggregate.html), as defined in Domain-Driven Design, using [Akka Persistence Typed](https://doc.akka.io/docs/akka/2.6/typed/persistence.html?language=java) and following the [[CQRS|ES_CQRS]] principles embraced by Lagom. While Akka Persistence Typed provides an API for building event-sourced actors, the same does not necessarily apply for CQRS Aggregates. To build CQRS applications, we need to use a few rules to our design.
+
+We use a simplified shopping cart example to guide you through the process. You can find a full-fledged shopping cart sample on our [samples repository](https://github.com/lagom/lagom-samples/tree/1.6.x/shopping-cart/shopping-cart-java).
+
+## Encoding the model
+
+First, create a class (ie: `ShoppingCartEntity`) extending the abstract class `EventSourcedBehaviorWithEnforcedReplies`.  It's recommended to use `EventSourcedBehaviorWithEnforcedReplies` when modeling a CQRS Aggregate. Using [enforced replies](https://doc.akka.io/docs/akka/2.6/typed/persistence.html?language=Java#replies) requires command handlers to return a `ReplyEffect` forcing the developers to be explicit about replies.
+
+@[shopping-cart-entity](code/docs/home/persistence/ShoppingCartEntity.java)
+
+`EventSourcedBehaviorWithEnforcedReplies` has three type parameters (Command, Event and State) and a few mandatory and optional methods that will be implemented as we progress through the guide.
+
+All model classes (`ShoppingCartEntity.Command`, `ShoppingCartEntity.Event`, `ShoppingCartEntity.ShoppingCart`) will be defined as static inner classes of `ShoppingCartEntity`.
+
+### Modelling the State
+
+The state of the shopping cart is defined as following:
+
+@[shopping-cart-state](code/docs/home/persistence/ShoppingCartEntity.java)
+
+The `ShoppingCart` has a `checkedOutTime` that can be set when transitioning from one state (open shopping cart) to another (checked-out shopping cart). As we will see later, each state encodes the commands it can handle, which events it can persist, and to which other states it can transition.
+
+> **Note**: This is not the recommended technique, though. Whenever your model goes through different state transitions, the recommendation is to have a trait and extensions of it for each state. See examples in the [style guide for Akka Persistence Typed](https://doc.akka.io/docs/akka/2.6/typed/persistence-style.html?language=Java).
```

Comment:
```suggestion
> **Note**: This is not the recommended technique, though. Whenever your model goes through different state transitions, the recommendation is to have an interface and implementations of it for each state. See examples in the [style guide for Akka Persistence Typed](https://doc.akka.io/docs/akka/2.6/typed/persistence-style.html?language=Java).
```

Reply:
As a user reading this for the first time I'd find "This is not the recommended technique" annoying: "So, here's how you do something, except this is not the way to do it, and you should do it that other way".

Instead I think we could rephrase to:

```suggestion
> **Note**: The sample shown above is a simplified case. Whenever your model goes through different state transitions, a better approach is to have an interface and implementations of it for each state. See examples in the [style guide for Akka Persistence Typed](https://doc.akka.io/docs/akka/2.6/typed/persistence-style.html?language=Java).
```